### PR TITLE
[DEV-1571] Feature - Newsroom Subscriptions SDK

### PR DESCRIPTION
**Pull request by M4rk07**
- Support Newsroom Subscriptions API in SDK.

### CODE REVIEW

File: src/Sdk/NewsroomSubscriptions/types.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Contact, UploadcareImage } from '../../types';
+
+export interface NewsroomSubscriptionCreateRequest {
+    email_address: string;
+    locale?: string;
+    url?: string | null;
```
Comment (kamilmielnik) [inline src/Sdk/NewsroomSubscriptions/types.ts:6]: Do `null` and `undefined` produce the same behavior? 
If yes, then I'd avoid using nulls.
If not, then it'd be nice to leave a comment explaining the difference.
This comment applies to all other similar cases.

**Review by kamilmielnik — APPROVED** [review]

File: src/Sdk/NewsroomSubscriptions/types.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ip_address?: string | null;
+    visitor_uid?: string | null;
+    session_uid?: string | null;
+    contact?: {
+        contact_type: Contact['contact_type'];
+        company_name: Contact['company_name'];
```
Comment (e1himself) [inline src/Sdk/NewsroomSubscriptions/types.ts:12]: `company_name` is only required when `contact_type === 'organisation'`

```suggestion
        company_name?: Contact['company_name'];
```

File: src/Sdk/NewsroomSubscriptions/types.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        first_name?: Contact['first_name'];
+        last_name?: Contact['last_name'];
+        gender?: Contact['gender'];
+        function_name?: Contact['function_name'];
+        organisations?: string[];
```
Comment (e1himself) [inline src/Sdk/NewsroomSubscriptions/types.ts:17]: `first_name`, `last_name`, `gender`, `function_name` and `organisations` is only available when `contact_type === "person". I'd mention this in properties comments.

File: src/Sdk/NewsroomSubscriptions/NewsroomSubscriptions.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import routing from '../routing';
+
+export default class NewsroomSubscriptions {
+    private readonly apiClient: ApiClient;
+
+    constructor({ apiClient }: { apiClient: ApiClient }) {
+        this.apiClient = apiClient;
+    }
+
+    public async create(
```
Comment (e1himself) [inline src/Sdk/NewsroomSubscriptions/NewsroomSubscriptions.ts:13]: I think it's worth to call this method `subscribe()`.

**Review by e1himself — COMMENTED** [review]
:+1:

File: src/Sdk/NewsroomSubscriptions/types.ts (hunk 1 above)
Comment (M4rk07) [reply src/Sdk/NewsroomSubscriptions/types.ts]: Yes, parameter not sent is the same as it was `null`, if that is what you asked? I removed the nulls. 👍

File: src/Sdk/NewsroomSubscriptions/types.ts (hunk 3 above)
Comment (M4rk07) [reply src/Sdk/NewsroomSubscriptions/types.ts]: Nice suggestion, I added comments. Thanks!

**Pull request merged**